### PR TITLE
Consistent unapply

### PR DESCRIPTION
**Pull request by otzzila**
This should switch transformers to using `withX()` operations. Doing this triggered some of our internal checks that were ignored previously. Transformers pass all tests, but it's a lot of code changed so I would appreciate a review.

### CODE REVIEW

**Comment by nancyday** [comment]
What happens if a sort defined by an enum is given a scope directly that does not match the # of enum elements?

**Comment by otzzila** [comment]
In that case an error is thrown, as before.

**Comment by nancyday** [comment]
AntiPrenexTransformer seems to be missing an update to the flags.

**Comment by nancyday** [comment]
> In that case an error is thrown, as before.

Can you document this in the code somewhere please?

File: core/src/main/scala/fortress/transformers/Quantifiers/QuantifiersToDefnsTransformer.scala
[the remark this answers is not in the document]
Comment (otzzila) [reply core/src/main/scala/fortress/transformers/Quantifiers/QuantifiersToDefnsTransformer.scala:1]: This transformer did not do a copy even, it constructed a new ProblemState from scratch, using most of the same values unchanged from before.

File: core/src/main/scala/fortress/transformers/Symmetry/SymmetryBreakingTransformer_MostUsed.scala
[the remark this answers is not in the document]
Comment (otzzila) [reply core/src/main/scala/fortress/transformers/Symmetry/SymmetryBreakingTransformer_MostUsed.scala:1]: Changed.

File: core/src/main/scala/fortress/transformers/Quantifiers/SkolemizeTransformer.scala
[the remark this answers is not in the document]
Comment (otzzila) [reply core/src/main/scala/fortress/transformers/Quantifiers/SkolemizeTransformer.scala:1]: Added.

**Review by ryandancy — APPROVED** [review]

**Comment by ryandancy** [comment]
Just pushed an update since there were a couple compile errors caused by remaining `withUnapplyInterp` calls.

**Pull request merged**